### PR TITLE
chore: update docs with release v13.0.0-rc2 changes

### PR DESCRIPTION
**Pull request by GAtom22**

### CODE REVIEW

**Review by Vvaradinov — COMMENTED** [review]
@GAtom22 We no longer need `DistributionAuthorization.sol`

File: docs/develop/smart-contracts/evm-extensions/authorization.md [hunk 1]
```diff
@@ -61,17 +72,34 @@ Find the [Solidity interface in the evmos/extensions repo](https://github.com/ev
...
-            string[] calldata methods
-        ) external returns (bool approved);
+        address spender,
+        string[] calldata methods,
+        string[] calldata allowedList
+    ) external returns (bool approved);
+    ```
+
+    The `allowedList` is the list of allowed addresses.
+    It always include the `tx.origin` if it is not passed in manually.
```
Comment (MalteHerrmann) [inline docs/develop/smart-contracts/evm-extensions/authorization.md:90]:
```suggestion
    It always includes the `tx.origin` if it is not passed in manually.
```

File: docs/develop/smart-contracts/evm-extensions/authorization.md [hunk 2]
```diff
@@ -61,17 +72,34 @@ Find the [Solidity interface in the evmos/extensions repo](https://github.com/ev
...
+    You can use this paramter on the `MsgSetWithdrawAddress` for example.
```
Comment (MalteHerrmann) [inline docs/develop/smart-contracts/evm-extensions/authorization.md:91]:
```suggestion
    You can use this parameter on the `MsgSetWithdrawAddress` for example.
```

File: docs/develop/smart-contracts/evm-extensions/authorization.md [hunk 3]
```diff
@@ -61,17 +72,34 @@ Find the [Solidity interface in the evmos/extensions repo](https://github.com/ev
         ) external returns (bool approved);
     ```
 
-### `GenericAuthorization.sol`
+### `DistributionAuthorization.sol`
 
 - `approve`
 
-    Approves a list of Cosmos message
+    Approves a list of Cosmos message.
```
Comment (MalteHerrmann) [inline docs/develop/smart-contracts/evm-extensions/authorization.md:79]: nit: definitely use plural, might make sense to also say transactions here or use messages down below to have coherence in the document.

```suggestion
    Approves a list of Cosmos transactions.
```

File: docs/develop/smart-contracts/evm-extensions/distribution.md [hunk 4]
```diff
@@ -265,13 +265,15 @@ string[] private distributionMethods = [
     MSG_WITHDRAW_VALIDATOR_COMMISSION,
 ];
 
-/// @dev Approves all distribution transactions.
+/// @dev Approves this smart contract to perform all distribution transactions on behalf of the transaction signer.
 /// @dev This creates a Cosmos Authorization Grant for the given methods.
-/// @dev This emits an Approval event from the GenericAuthorization.sol.
+/// @dev This emits an Approval event from the DistributionAuthorization.sol.
 function approveAllDistributionMethods() public {
+    string[] memory allowedList = new string[](0); // Will always add tx.origin address if it's not passed in manually
```
Comment (MalteHerrmann) [inline docs/develop/smart-contracts/evm-extensions/distribution.md:272]: nit
```suggestion
    string[] memory allowedList = new string[](0); // backend logic automatically adds tx.origin if it's not passed in manually
```

**Review by MalteHerrmann — APPROVED** [review]
LGTM, thanks for updating @GAtom22!

**Pull request merged**